### PR TITLE
Fix/model validation

### PR DESCRIPTION
**Pull request by esherman-credo**
## Describe your changes
Model validation: add support for pd.Series as a datatype for X
Predict_proba functionality: for custom (non-Dummy) classifiers, we're going to allow users to "signal" to Lens that we should treat their `predict` and `predict_proba` functions as being sklearn-like. If the user adds a `frameworks_like` attribute (a list) to their custom classifier and pass that classifier as the `model_like` argument to `ClassificationModel`, we will check the contents of `framework_like`. If `sklearn` is in the attribute, we will take steps similar to existing `framework` checking and functionality updating. Note that _multiple_ frameworks can be specified. For instance, a user might want to pass a pytorch model that supports both `forward()` (default pytorch behavior) and custom-implemented `predict` and `predict_proba` behavior a la sklearn. In this case, the user would set an attribute in their custom model `frameworks_like = ['sklearn', 'pytorch']`. Full pytorch support not yet implemented.

## Issue ticket number and link
Zoom discussion

## Known outstanding issues that are not fully accounted for
Full support of `frameworks_like` not yet implemented.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

**Review by fabrizio-credo — APPROVED** [review]

**Comment by amrasekh** [comment]
@esherman-credo  I have [this](https://github.com/credo-ai/customer_demos/blob/prod/prod/d11_sentiment_analysis/temporary/loan_classification.ipynb) model that throws error on this branch. 
The model is essentially an `sklearn` classification model that is wrapped to function based on a custom threshold (a very common scenario)
Is this PR envisioned to accommodate for it?
There is also [this](https://github.com/credo-ai/customer_demos/blob/prod/prod/d11_sentiment_analysis/sentiment_analysis.ipynb) model which is a similar, but more complicated one, that is throwing the same error as well.
Feel free to run the notebooks for tests, etc. Thanks!

**Review by amrasekh — APPROVED** [review]
Tested with a number of sklearn-like custom models and all work as desired.

File: credoai/lens/lens_validation.py [hunk 1]
```diff
@@ -125,8 +125,8 @@ def check_prediction_model_output(fn, data, batch: int = 1):
     batch_size = batch
```
Comment (fabrizio-credo) [inline credoai/lens/lens_validation.py:125]: Why the reassigning of batch?

File: credoai/lens/lens_validation.py [hunk 2]
```diff
@@ -125,8 +125,8 @@ def check_prediction_model_output(fn, data, batch: int = 1):
...
     if isinstance(data.X, np.ndarray):
         mini_pred = fn(np.reshape(data.X[0], (1, -1)))
-    elif isinstance(data.X, pd.DataFrame):
-        mini_pred = fn(data.X.head(1))
+    elif isinstance(data.X, (pd.DataFrame, pd.Series)):
+        mini_pred = fn(data.X.head(batch))
```
Comment (fabrizio-credo) [inline credoai/lens/lens_validation.py:129]: Should it be batch_size since you reassigned above?

**Review by fabrizio-credo — APPROVED** [review]
A minor comment, it looks OK to me.

File: credoai/lens/lens_validation.py (hunk 1 above)
Comment (esherman-credo) [reply credoai/lens/lens_validation.py:125]: I wanted to avoid confusion between input `batch` and output `batch_size` since its value could be changed depending on the data type of `X`.

This is a primarily internal function so I'm not sure it matters. I can change to batch_out and batch_in.
